Sources/AsyncHTTPClient/HTTPClient.swift

```diff
@@ -1012,6 +1035,84 @@ public final class HTTPClient: Sendable {
+            decompression: Decompression = .disabled,
+            http1_1ConnectionDebugInitializer: (@Sendable (Channel) -> EventLoopFuture<Void>)? = nil,
+            http2ConnectionDebugInitializer: (@Sendable (Channel) -> EventLoopFuture<Void>)? = nil,
+            http2StreamChannelDebugInitializer: (@Sendable (Channel) -> EventLoopFuture<Void>)? = nil,
+            tracing: TracingConfiguration = .init()
+        ) {
+            self.init(
+                tlsConfiguration: tlsConfiguration,
+                redirectConfiguration: redirectConfiguration,
+                timeout: timeout,
+                connectionPool: connectionPool,
+                proxy: proxy,
+                ignoreUncleanSSLShutdown: ignoreUncleanSSLShutdown,
+                decompression: decompression
+            )
+            self.http1_1ConnectionDebugInitializer = http1_1ConnectionDebugInitializer
+            self.http2ConnectionDebugInitializer = http2ConnectionDebugInitializer
+            self.http2StreamChannelDebugInitializer = http2StreamChannelDebugInitializer
+            self.tracing = tracing
+        }
+    }
+
+    public struct TracingConfiguration: Sendable {
+
+        @usableFromInline
+        var _tracer: Optional<any Sendable>  // erasure trick so we don't have to make Configuration @available
+
+        /// Tracer that should be used by the HTTPClient.
+        ///
+        /// This is selected at configuration creation time, and if no tracer is passed explicitly,
+        /// (including `nil` in order to disable traces), the default global bootstrapped tracer will
+        /// be stored in this property, and used for all subsequent requests made by this client.
+        @inlinable
+        @available(macOS 10.15, iOS 13, tvOS 13, watchOS 6, *)
+        public var tracer: (any Tracer)? {
+            get {
+                guard let _tracer else {
+                    return nil
+                }
+                return _tracer as! (any Tracer)?
+            }
+            set {
+                self._tracer = newValue
+            }
+        }
+
+        // TODO: Open up customization of keys we use?
+        /// Configuration for tracing attributes set by the HTTPClient.
+        @usableFromInline
+        package var attributeKeys: AttributeKeys
+
+        public init() {
+            self._tracer = nil
+            self.attributeKeys = .init()
+        }
+
+        /// Span attribute keys that the HTTPClient should set automatically.
+        /// This struct allows the configuration of the attribute names (keys) which will be used for the apropriate values.
+        @usableFromInline
+        package struct AttributeKeys: Sendable {
+            @usableFromInline package var requestMethod: String = "http.request.method"
+            @usableFromInline package var requestBodySize: String = "http.request.body.size"
+
+            @usableFromInline package var responseBodySize: String = "http.response.size"
```

Comment:
Is this missing a "body"? `requestBodySize` has one.

```suggestion
            @usableFromInline package var responseBodySize: String = "http.response.body.size"
```

Reply:
Good one, both actually exist it seems: 

- `http.response.body.size` - The size of the response payload body in bytes. This is the number of bytes transferred excluding headers and is often, but not always, present as the Content-Lengthheader. For requests using transport encoding, this should be the compressed size.
- `http.response.body.size` - The size of the response payload body in bytes. This is the number of bytes transferred excluding headers and is often, but not always, present as the [Content-Length](https://www.rfc-editor.org/rfc/rfc9110.html#field.content-length) header. For requests using transport encoding, this should be the compressed size.

but that's pending a whole discussion how far we wanna push the otel specific semantics here... 

Thanks, I'll fix this to be the body!

Reply:
Thank you! Actually I just realized that otel BOTH `http.response.size` and `http.response.body.size` exist, I guess I copied from the wrong row when checking these.

Let's make these only the body ones;